cadc-storage-adapter-fs/src/main/java/org/opencadc/inventory/storage/fs/OpaqueFileSystemStorageAdapter.java

```diff
@@ -238,20 +232,20 @@ private void init(Path root) {
                 log.debug("created txn dir: " + txnPath);
             }
             if (!Files.isReadable(txnPath) || (!Files.isWritable(txnPath))) {
-                throw new IllegalArgumentException("read-write permission required on transaction directory");
+                throw new InvalidConfigException("read-write permission required on transaction directory");
             }
             log.debug("validated txn dir: " + txnPath);
 
         } catch (InvalidPathException e) {
-            throw new IllegalArgumentException("Invalid root directory: " + root, e);
+            throw new InvalidConfigException("Invalid root directory: " + root, e);
         } catch (IOException io) {
-            throw new IllegalArgumentException(("Could not create content or transaction directory"), io);
+            throw new InvalidConfigException(("Could not create content or transaction directory"), io);
         }
     }
     
     @Override
     public void get(StorageLocation storageLocation, OutputStream dest)
-        throws ResourceNotFoundException, ReadException, WriteException, StorageEngageException, TransientException {
+        throws ResourceNotFoundException, ReadException, WriteException, WriteException, StorageEngageException, TransientException {
```

Comment:
Yeah, two is one too many :-)

WriteException is a little weird with get(...) but it's true: ReadException is from reading then StorageLocation bytes and WriteException is from writing to the OutputStream. 

The problem is really in the method signature: `void get` but I think @brianmajor and I chose it to match the minoc usage (which is the only place `get` is used outside of test code). It could be `void stream(StorageLocation storageLocation, OutputStream dest)` but not worth it right now... maybe in the pre-1.0 API review.